### PR TITLE
Make everything very tidy

### LINKED ISSUE #941: Fix clang-tidy issues
#0000 adds support for clang-tidy. By default, it is enabled in Jenkins' cmake-debug build. To enable it locally, run cmake with `-DENABLE_CLANG_TIDY=ON`.

When our naming schema is violated, it throws an error:

```
/Users/markus/Projekte/Opossum/Git/src/lib/operators/export_binary.cpp:24:6: error: invalid case style for function '_export_values' [readability-identifier-naming,-warnings-as-errors]
void _export_values(std::ofstream& ofstream, const std::vector<T, Alloc>& values);
     ^~~~~~~~~~~~~~~
     export_values

/Users/markus/Projekte/Opossum/Git/src/lib/operators/product.cpp:44:15: error: invalid case style for protected method 'add_product_of_two_chunks' [readability-identifier-naming,-warnings-as-errors]
void Product::add_product_of_two_chunks(std::shared_ptr<Table> output, ChunkID chunk_id_left, ChunkID chunk_id_right) {
              ^~~~~~~~~~~~~~~~~~~~~~~~~~
              _add_product_of_two_chunks
```

That's a start (and I have fixed those), but there are many, many more options that clang-tidy has. For instance, it tells us about a performance-related bug in aggregate.cpp:

```
/Users/markus/Projekte/Opossum/Git/src/lib/operators/aggregate.cpp:24:50: warning: the const qualified parameter 'groupby_column_ids' is copied for each invocation; consider making it a reference [performance-unnecessary-value-param]
                     const std::vector<ColumnID> groupby_column_ids)
                                                 ^
                                                &
```

Or about violations of the rule of three:
```
/Users/markus/Projekte/Opossum/Git/src/lib/operators/insert.cpp:19:7: warning: class 'AbstractTypedColumnProcessor' defines a default destructor but does not define a copy constructor, a copy assignment operator, a move constructor or a move assignment operator [cppcoreguidelines-special-member-functions]
class AbstractTypedColumnProcessor {
      ^
```

We should enable those checks that make sense and fix the related issues. I started a negative list of filters that don't make sense for us in `.clang-tidy:3`.

Small steps count as well - I do not expect to see a PR that fixes all issues at once.

At some point, we can hopefully turn most warnings into errors.

### PR DESCRIPTION
**Pull request by mrks**
Fixes clang-tidy's last complaints, enables all checks and sets them to raise errors. fixes #941

### CODE REVIEW

File: src/bin/server.cpp [hunk 1]
```diff
@@ -15,7 +16,10 @@ int main(int argc, char* argv[]) {
     uint16_t port = 5432;
 
     if (argc >= 2) {
-      port = static_cast<uint16_t>(std::atoi(argv[1]));
+      char* endptr{nullptr};
```
Comment (mrzzzrm) [inline src/bin/server.cpp]: `type_cast`?

Comment (mrks) [reply src/bin/server.cpp]: I think this version has a more robust error handling

Comment (mrzzzrm) [reply src/bin/server.cpp]: This version has unintended behaviour if the specified port is `>= 2 ^ 16` or less than zero ;) Don't know if type_cast (aka boost literal_cast) handles this better, though.

**Comment by mrks** [comment]
Hmm, looks like clang on Linux finds other issues than on OS X. oO

File: src/bin/server.cpp (hunk 1 above)
Comment (mrks) [reply src/bin/server.cpp]: Nope, doesn't. Will fix.

**Pull request merged**